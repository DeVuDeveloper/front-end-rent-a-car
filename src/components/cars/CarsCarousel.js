import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import Paginate from './Paginate';
import insta from '../../assets/images/insta-icon-home.png';
import twitter from '../../assets/images/twitter-icon-home.png';
import facebook from '../../assets/images/fb-icon-home.png';

const social = [
  { icon: facebook },
  { icon: twitter },
  { icon: insta },
];

const CarsCarousel = () => {
  const cars = useSelector((state) => state.cars);
  const [pageNumber, setPageNumber] = useState(0);
  const carsPerPage = 3;
  const pageVisited = pageNumber * carsPerPage;
  const displayCars = cars.slice(pageVisited, pageVisited + carsPerPage);
  const pageCount = Math.ceil(cars.length / carsPerPage);

  const changePage = ({ selected }) => {
    setPageNumber(selected);
  };

  return (
    <div className="flex flex-col gap-4 mt-16 lg:flex-row">
      {displayCars.map((item) => (
        <div className="flex flex-col items-center w-80" key={item.id}>
          <Link to={`/CarDetails/${item.id}`}>
            <img src={item.carImg} alt="car" className="car rounded-full w-60 h-60 m-auto blob" />
            <h2 className="text-center">{item.carModel}</h2>
            <hr className="border-t-2 border-dashed border-gray-100 w-32 mt-4 m-auto" />
            <p className="text-gray-400 text-center font-semibold mt-4">
              {item.carDescription}
            </p>
          </Link>
          <div className="flex gap-4 mt-2">
            {social.map((item) => (
              <a href="/" key={item.icon}><img src={item.icon} alt="social" className="w-8" /></a>
            ))}
          </div>
        </div>
      ))}
      <Paginate pageCount={pageCount} changePage={changePage} />
    </div>
  );
};

export default CarsCarousel;
