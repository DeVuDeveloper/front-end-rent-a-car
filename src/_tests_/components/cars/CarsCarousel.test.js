import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import CarsCarousel from '../../../components/cars/CarsCarousel';
import store from '../../../redux/configureStore';

it('should render CarCarousel component correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <CarsCarousel />
      </Router>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
