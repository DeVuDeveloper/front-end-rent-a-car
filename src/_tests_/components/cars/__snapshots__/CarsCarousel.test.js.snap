// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CarCarousel component correctly 1`] = `
<div
  className="flex flex-col gap-4 mt-16 lg:flex-row"
>
  <ul>
    <li
      className="previous disabled"
    >
      <a
        aria-disabled="true"
        aria-label="Previous page"
        className="previous-btn disable-btns"
        onClick={[Function]}
        onKeyPress={[Function]}
        rel="prev"
        role="button"
        tabIndex="-1"
      >
        ᐊ
      </a>
    </li>
    <li
      className="next"
    >
      <a
        aria-disabled="false"
        aria-label="Next page"
        className="next-btn"
        onClick={[Function]}
        onKeyPress={[Function]}
        rel="next"
        role="button"
        tabIndex="0"
      >
        ᐅ
      </a>
    </li>
  </ul>
</div>
`;
